crates/bevy_asset/src/asset_server.rs

```diff
@@ -515,23 +525,36 @@ impl AssetServer {
     }
 }
 
-pub fn free_unused_assets_system(asset_server: Res<AssetServer>) {
-    asset_server.free_unused_assets();
+fn free_unused_assets_system_ref(asset_server: &AssetServer) {
+    // note: order of these function is important.
```

Comment:
it was ordered the other way to introduce a one frame delay between marking an asset as unused and actually deleting it

Reply:
Ohh, so it was intentional? 
Ok good to know. I'll adjusts the tests to reflect that behavior.